### PR TITLE
fix: forest coverity issues

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
# Description
Pr contains fixes for coverity in Decision Forest algorithm

### CODE REVIEW

File: cpp/daal/src/algorithms/dtrees/dtrees_train_data_helper.i [hunk 1]
```diff
@@ -119,7 +119,8 @@ struct SplitData
           nLeft(0),
           iStart(0),
           totalWeights(0.0),
-          leftWeights(0.0)
+          leftWeights(0.0),
+          left {}
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/dtrees/dtrees_train_data_helper.i:123]: Isn't it best practice to instantiate in the same order we declare in the class?

**Comment by Alexsandruss** [comment]
@mergifyio rebase

**Review by Alexsandruss — CHANGES_REQUESTED** [review]
Correct args order for SplitData

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**